mcf-browser/src/back-end/parse-mcf.js

```diff
@@ -0,0 +1,258 @@
+   * of syntax error in the mcf file.
+   * @type {number}
+   */
+
+  lineNum;
+  /**
+   * Create a ParseMcf object which keeps tracks of the current source node of
+   * each triple in the mcf and the provenance, which is the mcf file name.
+   * @param {string} fileFullName Name of file to be converted to a provenance
+   *     representing the file in teh Assertion Objects created while parsing.
+   */
+  constructor(fileFullName) {
+    let fileName = fileFullName;
+
+    // remove file extension to keep only filename as provenance
+    if (fileFullName.includes('.')) {
+      const fileNameSpiltOnPeriod = fileFullName.split('.');
+      fileNameSpiltOnPeriod.pop();
+      fileName = fileNameSpiltOnPeriod.join('.');
+    }
+    this.prov = 'local: ' + fileName;
+    this.curNode = null;
+    this.lineNum = -1;
+  }
+
+  /**
+   * Parses a string represetnaing a comma separated list of property values
+   * from a line of an mcf file. Returns a list having either a string or
+   * <namespace, reference> pair.
+   *
+   * @param {string} propValues A comma separated list of property values.
+   * @return {Array<(string|Object)>} Array of
+   *     parsed values.
+   */
+  parsePropValues(propValues) {
+    if (!propValues) {
+      throw new Error('Error, no property values to parse (line ' +
+                      this.lineNum + ')');
+    }
+    const values = [];
+    for (const propValue of propValues.split(',')) {
+      if (propValue.split(':').length === 2) {
+        const namespace = propValue.split(':')[0].trim();
+        if (NAMESPACES.indexOf(namespace) >= 0) {
+          values.push({
+            'ns' : namespace,
+            'ref' : propValue.split(':')[1].trim(),
+          });
+          continue;
+        }
+      }
+      values.push(propValue.trim());
+    }
+    return values;
+  }
+  /**
+   * Sets curNode variable of the calling ParseMcf object based on the passed in
+   * parsed values of a line of mcf, given the property label for the line was
+   * 'Node'. The parsed value either is a local reference with or without the
+   * 'l' namespace, or it has a remote namespace. If the namespace is remote,
+   * then the dcid for curNode is set. Updates localNodeHash mapping to store
+   * the subject nodes to be displayed in home screen of browser.
+   *
+   * @param {Array<string|Object>} parsedValues The array of parsed values from
+   *     a line of mcf with property label of 'Node'.
+   */
+  setCurNode(parsedValues) {
+    if (parsedValues.length !== 1) {
+      throw new Error('Error in declaring node (line ' + this.lineNum +
+                      '): ' + parsedValues.toString());
+    }
+
+    let isDCID = false;
+    let nodeId = parsedValues[0];
+
+    if (parsedValues[0] instanceof Object) {
+      nodeId = parsedValues[0]['ref'];
+      if (parsedValues[0]['ns'] !== LOCAL_NS) {
+        isDCID = true;
+      }
+    }
+
+    this.curNode = Node.getNode(nodeId, /* shouldCreate */ true);
```

Comment:
1. In the main parser, for `Node: dcid:foo`, the local-id is `dcid:foo` and DCID is `foo`.
2. We want to treat only `dcid:` prefix in this case (it is a short-form of defining the dcid property, rather than the namespace).
3. Finally, any other namespace prefix (l, dcs, schema) would be invalid.

Considering all of the above, I might suggest something like this:

```
let dcid = '';

if (parsedValues[0] instanceof Object) {
  let ns = parsedValues[0]['ns'];
  if (ns === 'dcid') {
    dcid = parsedValues[0]['ref'];
  } else if (NAMESPACES.includes(ns)) {
    // throw error
  }
}

this.curNode = Node.get(nodeId, true);
if (dcid !== '') {
  this.curNode.setDCID(dcid);
}
```

Reply:
when the main parser uses 'dcid:foo' as the local-id for the node, does that mean that other nodes could refer to the node with 'l:dcid:foo' ? Or would other nodes solely use the dcid to link to foo?